pkg/framework/machinesets.go

```
@@ -141,12 +144,86 @@ func buildMachineSetParamsFromMachineSet(ctx context.Context, client runtimeclie
+// only be used as a template: it must not be treated as a live object with real status or
+// owned Machines.
+//
+// Returns errNoWorkerMachineSetsFound when neither a MAPI nor a CAPI worker MachineSet could
+// be found.
+func GetSampleMAPIWorkerMachineSet(ctx context.Context, client runtimeclient.Client) (*machinev1.MachineSet, error) {
 	workers, err := GetWorkerMachineSets(ctx, client)
-	Expect(err).ToNot(HaveOccurred(), "listing Worker MachineSets should not error.")
+	if err != nil {
+		return nil, fmt.Errorf("error listing worker MachineSets: %w", err)
+	}
+	if len(workers) > 0 {
```

golangci-lint (2.12.2)

[error] missing whitespace above this line (invalid statement above if)

(wsl_v5)